xdsl/irdl/attributes.py

```diff
@@ -431,3 +432,9 @@ def irdl_to_attr_constraint(
         )
 
     raise ValueError(f"Unexpected irdl constraint: {irdl}")
+
+
+def constr(
+    irdl: GenericAttrConstraint[AttributeInvT] | AttributeInvT | type[AttributeInvT],
+) -> GenericAttrConstraint[AttributeInvT]:
+    return cast(GenericAttrConstraint[AttributeInvT], irdl_to_attr_constraint(irdl))
```

Comment:
can I has docstring to explain why it's necessary?

Reply:
Added docstrings